### ISSUE TITLE
`range(b, e, step)` functions

### ISSUE THREAD

**Issue by JonathanStarup** [flix/flix#6997]

**Comment by LionelMeli** [comment]
Hi Jonathan,

I'm interested by this issue (something like `Iterator.range(rc: Region[r], b: Int32, e: Int32, step: Int32): Iterator[Int32, r, r] \ r` )

**Comment by magnus-madsen** [comment]
I think we should be looking at adding a Range data type.

**Comment by JonathanStarup** [comment]
Yeah I think i was going down a wrong path by talking about iterator, it should really be on list instead, or a range data type
